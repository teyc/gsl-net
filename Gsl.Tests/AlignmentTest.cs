using Xunit;
using Xunit.Abstractions;

namespace Gsl.Tests
{
    public class AlignmentTest: TestBase
    {
        public AlignmentTest(ITestOutputHelper log): base(log) { }

        [Fact]
        public void AlignProperties()
        {
            const string template = "data/align.gsl";
            const string data = "data/align.json";

            TemplateWithSingleFileOutput(template, data);
        }

        [Fact]
        public void NestedAlignProperties()
        {
            const string template = "data/align2.gsl";
            const string data = "data/align2.json";

            TemplateWithSingleFileOutput(template, data);
        }

        [Fact]
        public void Optional()
        {
            const string template = "data/optional.gsl";
            const string data = "data/optional.json";

            TemplateWithSingleFileOutput(template, data);
        }
    }
}
