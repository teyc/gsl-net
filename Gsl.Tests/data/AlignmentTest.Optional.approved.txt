CREATE TABLE foo (
    FirstName INTEGER NOT NULL,
    LastName INTEGER NOT NULL,
    DateOfBirth INTEGER NOT NULL
)